transport-native-kqueue/src/main/c/netty_kqueue_bsdsocket.c

```diff
@@ -159,8 +159,16 @@ static jobject netty_kqueue_bsdsocket_getPeerCredentials(JNIEnv *env, jclass cla
         (*env)->SetIntArrayRegion(env, gids, 0, 1, (jint*) &credentials.cr_gid);
     }
 
-    // TODO: getting the PID may require reading/sending "ancillary data" via SCM_CREDENTIALS which is not desirable.
-    return (*env)->NewObject(env, peerCredentialsClass, peerCredentialsMethodId, 0, credentials.cr_uid, gids);
+    pid_t pid = 0;
+#ifdef LOCAL_PEERPID
+    socklen_t len = sizeof(pid);
+    // Getting the LOCAL_PEERPID is expected to return error in some cases (e.g. server socket FDs) - just return 0.
+    if (netty_unix_socket_getOption0(fd, SOCK_STREAM, LOCAL_PEERPID, &pid, len) < 0) {
```

Comment:
why do we need to use netty_unix_socket_getOption0 here? could we use netty_unix_socket_getOption as well and skip assigning len to a variable to keep it consistent with the style above.

Reply:
netty_unix_socket_getOption will raise an exception for error return values, and as commented, we do not want an exception in case of server file descriptors.

Reply:
`netty_unix_socket_getOption` will produce an exception, while `netty_unix_socket_getOption0` will not. So I think it is correct to use `netty_unix_socket_getOption0`.